src/Engine/Animator/Animator.cpp

```diff
@@ -0,0 +1,70 @@
+#include "Animator.h"
+
+Animator::Animator()
+    : initialized_(false)
+{}
+
+// Keeping this as an init right now because the objects may not
+int Animator::init(SDL_Texture* texture, int rows, int cols, int period) {
+    period_ = period;
+    rows_ = rows;
+    cols_ = cols;
+    current_index_ = 0;
+
+    //If there was no draw_order given as an argument, we will create a default
+    if (draw_order_.size() == 0) {
+        for (int i = 0; i < rows * cols; i++) {
+            draw_order_.push_back(i);
+        }
+    }
+
+    // We will run off the assumption that the given
+    // spritesheet has it's starting sprites on 0,0.
+    // If there is padding... manually get rid of it for now
```

Comment:
This might imply manually getting rid of it in code, I think it should be clearer that this means manually remove padding from the pictures using external tools.